fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreTest.java

```diff
@@ -2554,4 +2554,56 @@ public void invalidMetaData() throws Exception {
             assertThrows(KeyExpression.InvalidExpressionException.class, () -> openSimpleRecordStore(context, invalid));
         }
     }
+
+    @Test
+    public void commitChecks() throws Exception {
+        // Start check now; fails even if added.
+        try (FDBRecordContext context = openContext()) {
+            openSimpleRecordStore(context);
+
+            context.addCommitCheck(checkRec1Exists());
+
+            TestRecords1Proto.MySimpleRecord rec = TestRecords1Proto.MySimpleRecord.newBuilder()
+                    .setRecNo(1L)
+                    .build();
+            recordStore.saveRecord(rec);
```

Comment:
I'm trying to reason about whether there is a race condition here or not. Both checking if record 1 exists and adding the record end up reading the same key, so I *think* adding the record will always be serialized after the read on the network thread, so I guess not. Still seems maybe a little delicate.

Would something like:

```java
CompletableFuture<Void> check = checkIfRec1Exists());
context.addCommitCheck(check);
check.exceptionally(eignore -> null).get();
```

work to accomplish the same thing? It means the check has actually completed before the record save begins, which maybe means the test loses something.

Reply:
I actually thought RYW would even serialize a read before a blind write to the same key. I realize that this test ends up being one of that sort of behavior, but changing it to do things explicitly really means not using a commit check at all, it feels to me.

Reply:
Okay, I can live with that. I think you're right, so I suppose just waiting for flakiness from this test is fine (if it turns out that things aren't serialized that way in the network thread).